File: Python/009.py

def Pythagorean(a, b, c):
    if a ** 2 + b ** 2 == c ** 2:
        return True
    else:
        return False

def solve():
    for a in range(1, 1000):
        for b in range(a + 1, 1000):
            c = 1000 - a - b
            if Pythagorean(a, b, c):
                return a, b, c
    return None

a, b, c = solve()
print(a * b * c)

File: C++/009.cpp
#include <iostream>
using namespace std;

int solve(int n)
{
    int a, b, c;
    for (a = 1; a < n; a++)
        for (b = a + 1; b < n; b++) {
            c = n - a - b;
            if (a * a + b * b == c * c)
                return a * b * c;
        }
    return 0;
}

int main()
{
    int n = 1000;
    cout << solve(n) << endl;
}